public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -347,6 +404,134 @@ export function GraphiteDatasource(instanceSettings, $q, backendSrv, templateSrv
       });
   };
 
+  this.createFuncInstance = function(funcDef, options?) {
+    return gfunc.createFuncInstance(funcDef, options, this.funcDefs);
+  };
+
+  this.getFuncDef = function(name) {
+    return gfunc.getFuncDef(name, this.funcDefs);
+  };
+
+  this.getFuncDefs = function() {
```

Comment:
this function is quite long and does quite a bit, would be nice with some unit tests for the response handling of the function definitions.